cpp/cycles_module/cycles_test.cpp

```diff
@@ -0,0 +1,285 @@
+#include <algorithm>
+#include <chrono>
+#include <random>
+
+#include <gtest/gtest.h>
+#include <mg_generate.hpp>
+#include <mg_graph.hpp>
+
+#include "algorithm/cycles.hpp"
+
+namespace {
+bool CheckCycles(std::vector<std::vector<mg_graph::Node<>>> user, std::vector<std::vector<std::uint64_t>> correct) {
+  // normalize cycles
+  for (auto &cycle : correct) {
+    std::rotate(cycle.begin(), std::min_element(cycle.begin(), cycle.end()), cycle.end());
+  }
+
+  std::vector<std::vector<std::uint64_t>> user_cycles;
+  for (const auto &cycle : user) {
+    std::vector<std::uint64_t> user_cycle;
+    for (const auto &node : cycle) {
+      user_cycle.push_back(node.id);
+    }
+    std::rotate(user_cycle.begin(), std::min_element(user_cycle.begin(), user_cycle.end()), user_cycle.end());
+    user_cycles.push_back(user_cycle);
+  }
+
+  if (user_cycles.size() != correct.size()) {
+    return false;
+  }
+
+  for (size_t i = 0; i < user_cycles.size(); ++i) {
+    user_cycles[i].push_back(*user_cycles[i].begin());
```

Comment:
```suggestion
    asssert(user_cycles[i].size() > 0);
    user_cycles[i].push_back(user_cycles[i][0]);
```
or just 
```suggestion
    user_cycles[i].push_back(user_cycles[i][0]);
```
since these are tests...